admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -642,6 +645,25 @@ var specificPrices = (function() {
           $('#form_step2_specific_price_sp_reduction_tax').show();
         }
       });
+
+      /* enable price field only when needed */
+      leaveInitialPrice.on('click', function togglePriceField() {
+        productPriceField.prop('disabled', $(this).is(':checked'))
+          .val('')
+        ;
+      });
+
+      /* enable tax type field only when reduction by amount is selected */
+      discountTypeField.on('change', function toggleDiscountTaxField() {
+        var uglySelect2Selector = $('#select2-form_step2_specific_price_sp_reduction_tax-container').parent().parent();
```

Comment:
I tried a lot of selectors, without any success. This block is generated buy select2 plugin, and it's realy a pain to interact with :/

I'm open to any improvement of this selector, though (this is why I use a variable and name it this way).